app/services/course/assessment/submission/update_service.rb

```diff
@@ -126,8 +126,9 @@ def create_missing_submission_questions
   def update_submission
     @submission.class.transaction do
       update_answers_params[:answers]&.each do |answer_params|
-        answer = @submission.answers.detect { |answer| answer.id == answer_params[:id].to_i }
-        update_answer(answer, answer_params)
+        next if answer_params[:id].blank?
+        answer = @submission.answers.includes(:actable).detect { |a| a.id == answer_params[:id].to_i }
```

Comment:
In order to differentiate `Array#find`  and `ActiveRecord::Base.find`, think we should use `detect`. `Array#detect` is the alias of `Array#find`